doc/specs/Proto extensions-spec.md

```diff
@@ -0,0 +1,378 @@
+        "brightPurple" : "#B4009E",
+        "brightRed" : "#E74856",
+        "brightWhite" : "#F2F2F2",
+        "brightYellow" : "#F9F1A5"
+    }
+  ]
+}
+```
+
+### Creation and location(s) of the json files
+
+#### For apps installed through Microsoft store (or similar)
+
+For apps that are installed through something like the Microsoft Store, they will need to declare themselves to
+be an app extension or write a separate app extension. In the app extension, they will need to create a public
+folder and insert their json files into it. The specifics of how they can do this are provided in the
+[Microsoft Docs](https://docs.microsoft.com/en-us/windows/uwp/launch-resume/how-to-create-an-extension) for how
+to create an app extension, and I will replicate the necessary section here. 
+
+In the appxmanifest file of the package:
+
+```
+<Package
+  ...
+  xmlns:uap3="http://schemas.microsoft.com/appx/manifest/uap/windows10/3"
+  IgnorableNamespaces="uap uap3 mp">
+  ...
+    <Applications>
+      <Application Id="App" ... >
+        ...
+        <Extensions>
+          ...
+          <uap3:Extension Category="windows.appExtension">
+            <uap3:AppExtension Name="Microsoft.com.Terminal"
+                               Id="<id>"
+                               PublicFolder="Public">
+            </uap3:AppExtension>
+          </uap3:Extension>
+        </Extensions>
+      </Application>
+    </Applications>
+    ...
+</Package>
+```
+
+Note that the name field **must** be `Microsoft.com.Terminal` for us to detect this extension. The `Id` field
+can be filled out as the app wishes. The `PublicFolder` field should have the name of the folder, relative to
+the package root, where the `json` files they wish to share with us are stored (this folder is typically named
+`Public` but can be renamed as long as it matches the relevant folder).
+
+During our profile generation, we will probe the OS for app extensions with the name `Microsoft.com.Terminal`
+and obtain the json files stored in the public folders of those app extensions.
+
+#### For apps installed 'traditionally' 
+
+For apps that are installed 'traditionally', there are 2 cases. The first is that this installation is for all
+the users of the system - in this case, the installer should add their json files to the global folder:
+
+```
+C:\ProgramData\Microsoft\Windows Terminal\Fragments
+```
+
+Note: `C:\ProgramData` is a [known folder](https://docs.microsoft.com/en-us/previous-versions/windows/desktop/legacy/bb776911(v=vs.85))
+that apps should be able to access. It will be on us to create the folder `Windows Terminal` in `C:\ProgramData\Microsoft\Windows`
+for this to happen. 
+
+In the second case, the installation is only for the current user. For this case, the installer should add the
+json files to the local folder:
+
+```
+C:\Users\<user>\AppData\Local\Microsoft\Windows Terminal\Fragments
+```
+
+We will look through both folders mentioned above during profile generation. 

```

Comment:
Right now the order is `local` -> `global` -> `app`. Though if we eventually go for something like an 'altitudes' system then it wouldn't matter which directory is scanned first since it will be about the json stub itself.